index.rb

```diff
@@ -0,0 +1,42 @@
+def bubble_sort(arr)
+  run = arr.length - 2
+  arr.each do
+    arr.each_with_index do |_, index|
+      next if arr[index + 1].nil? == true
+
+      next if arr[index] < arr[index + 1]
+
+      temp = arr[index]
+      arr[index] = arr[index + 1]
+      arr[index + 1] = temp
+    end
+    run -= 1
+  end
+  arr
+end
+
+def bubble_sort_by(arr)
+  run = arr.length - 2
+  arr.each do
+    arr.each_with_index do |_, index|
+      next if arr[index + 1].nil? == true
+
+      next if arr[index].length < arr[index + 1].length
+
+      temp = arr[index]
+      arr[index] = arr[index + 1]
+      arr[index + 1] = temp
+    end
+    run -= 1
+  end
+  arr
+end
```

Comment:
Please make use of the `yield` keyword to invoke the block passed to this method  ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#assignment-1)).